storm-core/src/jvm/backtype/storm/utils/TransferDrainer.java

```diff
@@ -30,18 +31,20 @@
   private HashMap<Integer, ArrayList<ArrayList<TaskMessage>>> bundles = new HashMap();
   
   public void add(HashMap<Integer, ArrayList<TaskMessage>> taskTupleSetMap) {
-    for (Integer task : taskTupleSetMap.keySet()) {
-      addListRefToMap(this.bundles, task, taskTupleSetMap.get(task));
+    for (Map.Entry<Integer, ArrayList<TaskMessage>> entry : taskTupleSetMap.entrySet()) {
+      Integer task = entry.getKey();
+      addListRefToMap(this.bundles, task, entry.getValue());
```

Comment:
Perhaps this could be simplified to  addListRefToMap(this.bundles, entry.getKey(), entry.getValue());